junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/Root.java

```diff
@@ -81,12 +98,22 @@ void prune() {
 		acceptInAllTestEngines(TestDescriptor::prune);
 	}
 
-	private boolean isExcluded(TestDescriptor descriptor, Filter<TestDescriptor> postDiscoveryFilter) {
-		return descriptor.getChildren().isEmpty() && postDiscoveryFilter.apply(descriptor).excluded();
+	private boolean isExcluded(TestDescriptor descriptor, FilterResult filterResult) {
+
+		return descriptor.getChildren().isEmpty() && filterResult.excluded();
 	}
 
 	private void acceptInAllTestEngines(TestDescriptor.Visitor visitor) {
 		this.testEngineDescriptors.values().forEach(descriptor -> descriptor.accept(visitor));
 	}
 
+	/**
+	 * prints the reason for test cases that are excluded in post discovery filter phase.
+	 */
+	private void printFilteredReason() {
```

Comment:
Could you add an info log here saying that `x container and y tests were excluded because <reason>`?
The `TestDescriptor` has `isContainer()` and `isTest()` methods you can use.
Also, the reason would have to be changed to something like `tags (do not) match the tag expression <tagExpression>` so that it makes a valid sentence.